workshop/content/lab1.6_operator.md

```diff
@@ -0,0 +1,179 @@
+
+1. Deployment configuration
+2. New immutable revision of the service
+3. Route
+4. Service
+5. Horizontal Pod Autoscaler
+
+For a developer, yaml might not be their preferred method of deployment, but for operations embracing the notion of gitops, a yaml deployment configuration that can be version controlled and tracked is the ideal artifact for their workflow.
+
+## Deployment
+
+First, make sure you're in the right project.
+
+```
+# make sure USER_NUMBER is set
+# export USER_NUMBER=1000
+oc project user-$USER_NUMBER
+```
+
+Now let's deploy our `hello-python` app from earlier, this time using a yaml file.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/deployment/hello-python.yml | oc apply -f -
+```
+
+Here's how that file looks, with comments added.
+
+```
+# Knative service
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  # name of the service
+  name: hello-python
+spec:
+  template:
+    metadata:
+      # optional name of our revision
+      name: hello-python-v1
+    spec:
+      containers:
+        # image that will be deployed
+        - image: quay.io/jkeam/hello-python:latest
+```
+
+Let's verify that our application was deployed.
+
+```
+oc get ksvc
+```
+
+We can also see the public route (URL).
+
+```
+NAME           URL                                                                                      LATESTCREATED     LATESTREADY       READY   REASON
+hello-python   http://hello-python-user-1.apps.cluster-tysons-5b24.tysons-5b24.sandbox891.opentlc.com   hello-python-v1   hello-python-v1   True
+```
+
+Open the URL in your browser and you should see
+
+```
+Hello World!
+```
+
+## Deployment v2
+Let's imagine now that the development team has delivered a new version of the `hello-python` application.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/deployment/hello-python-v2.yml | oc apply -f -
+```
+
+Here's that new deployment file, with comments added.
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello-python
+spec:
+  template:
+    metadata:
+      # new name of our revision
+      name: hello-python-v2
+    spec:
+      containers:
+      containers:
+        - image: quay.io/jkeam/hello-python
+          # new environment variable applied to this configuration so we can tell the difference between our service versions
+          env:
+            - name: TARGET
+              value: "World v2"
+```
+
+Now refresh the browser and you should see `Hello World v2`.
+
+## Blue-Green Deployment
+
+Deploying a new Knative service actually creates a new immutable revision and routes all traffic to it.  These immutable revisions allows us to have sophisticated deployment strategies.  The first we will examine is blue-green.
+
```

Comment:
Can we have something like to read more about what a blue-green deployment is, check out this link:
https://martinfowler.com/bliki/BlueGreenDeployment.html

Reply:
Great idea, will add.

Reply:
Just added.